### PR TITLE
LPS-176918 SF

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:stop

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
In Review :)

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -1182,6 +1183,14 @@ private ObjectDefinition _publishObjectDefinition(
 			throw new RequiredObjectFieldException();
 		}
 
+		count = _objectFieldPersistence.countByODI_L(
+			objectDefinition.getObjectDefinitionId(), true);
+
+		if (!objectDefinition.isEnableLocalization() && (count > 0)) {
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:1189]: @MarinhoFeliphe I believe this validation should be made while creating localized fields.

For example, the value of the `enableLocalization` only turn into `false` when there's no localized fields (while creating/updating ObjectDefinition). If we guarantee this, we don't need this check here.

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Hey @MarinhoFeliphe! Could you explain me better que last commit we have? Couldn't get it.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (MarinhoFeliphe) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: @gabrielwas agreed!

done [here](https://github.com/liferay-objects/liferay-portal/pull/1836/commits/95bb60c3d788f1f9a903dece61af027abb195456)

thank you!

**Comment by MarinhoFeliphe** [comment]
@gabrielwas 

About the last commit, I've fixed in [this commit](https://github.com/liferay-objects/liferay-portal/pull/1836/commits/0b5e7f78810f858c9bf25aaa5fafab6acc78b68b)

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 2]
```diff
@@ -1066,6 +1069,36 @@ private void _validateListTypeDefinitionId(
 		}
 	}
 
+	private void _validateLocalized(
+			String businessType, boolean localized,
+			ObjectDefinition objectDefinition)
+		throws PortalException {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPS-146755") || !localized) {
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:1081]: @MarinhoFeliphe is it possible here the following scenario:The FF is not enabled (so, this method will be returned) and we have the flag localized true?

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Just a question

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java (hunk 2 above)
Comment (MarinhoFeliphe) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java]: @gabrielwas I've just changed it [here](https://github.com/liferay-objects/liferay-portal/pull/1836/commits/769bc69abb912a45df9422dc3fe8695ca6ed1e25)

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Review by gabrielwas — APPROVED** [review]
LGTM :) Thx @MarinhoFeliphe !

**Comment by gabrielwas** [comment]
We can ask QAs analysis now

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/132429

**Pull request closed without merging**